File: src/206. Concealed Square/206.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Main {

    public static void main(String[] args) {

        Pattern r = Pattern.compile("1\\d2\\d3\\d4\\d5\\d6\\d7\\d8\\d9");
        for (long n = (long) Math.sqrt(10203040506070809.0); n < (long) Math.sqrt(19293949596979899.0); n++) {
            if (n % 10 == 3 || n % 10 == 7) {
                // note that n^2 is within the range of 64-bit integer, i.e, long
                String s = Long.toString(n * n);
                Matcher m = r.matcher(s);
                if (m.find()) {
                    System.out.println(n * 10);
                    break;
                }
            }
        }
    }

}

File: src/206. Concealed Square/206.go
package main

import (
	"fmt"
	"math"
	"regexp"
	"strconv"
)

func main() {
	r, _ := regexp.Compile("1\\d2\\d3\\d4\\d5\\d6\\d7\\d8\\d9")
	for i := int64(math.Sqrt(10203040506070809)); i < int64(math.Sqrt(19293949596979899)); i++ {
		if i % 10 == 3 || i % 10 == 7 {
			s := strconv.FormatInt(i * i, 10)
			m := r.MatchString(s)
			if m {
				fmt.Println(i * 10)
				break
			}
		}
	}
}
